### PR TITLE
Fix show_previews regression introduced in 2.73.0

### LINKED ISSUE #1541: BUG: Version `2.73.0` ignores  explicitly set`options.show_previews`
>**Note**
> I was only able to discover this as I work on internal application that shows component previews in production

[This particular commit](https://github.com/viewcomponent/view_component/compare/v2.72.0...v2.73.0#diff-fe26dc052312860cc076b99a7f3c0913bf834fc73fe246def378024ffc038164R22) that shipped with version `2.73.0` removes an important condition when setting the value of `options.show_previews`: 

Before: 

```
options.show_previews = Rails.env.development? || Rails.env.test? if options.show_previews.nil?
```

After: 

```
options.show_previews = Rails.env.development? || Rails.env.test?
```

That leads to the value of `options.show_previews` set in the environment config files to be ignored (e.g., a user might want to set ` config.view_component.show_previews = false` for development or `true` for production). 

This leads to further problems down the line like the `engine.rb` not being able to add the custom folder for previews to the autoload paths 

https://github.com/ViewComponent/view_component/blob/e34f08498e2d5dce8ac036decf1e719443ceb396/lib/view_component/engine.rb#L53

This is a clear regression and I will open a PR referencing this issue with a one-line fix, hoping the new gem version can be released ASAP.

### PR DESCRIPTION
**Pull request by progapandist**
Fixes #1541

### CODE REVIEW

**Review by Spone — APPROVED** [review]
Thanks Andy!

**Comment by Spone** [comment]
Can you just add a Changelog entry?

**Comment by progapandist** [comment]
Thanks for being reactive, Hans! Will fix the linter and bump the changelog right away

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ nav_order: 5
 
     *Joel Hawksley*
 
+* Fix `show_previews` regression introcuded in 2.73.0.
```
[the remark this answers is not in the document]
Comment (progapandist) [reply docs/CHANGELOG.md:21]: 🤦

**Pull request merged**